docs/drafts/derive-serviceaccount.md

```diff
@@ -0,0 +1,578 @@
+# Derive minimal ServiceAccount required for ClusterExtension Installation and Management
+
+OLM v1 does not have permission to install extensions on a cluster by default. In order to install a [supported bundle](../refs/supported-extensions.md), OLM must be provided a ServiceAccount configured with the appropriate permissions. For more information, see the [provided ServiceAccount](./provided-serviceaccount.md) documentation.
+
+This document serves as a guide for how to derive the RBAC necessary to install a bundle.
+
+### Required RBAC
+
+You can determine the specifics of these permissions by referencing the bundle of the ClusterExtension you want to install. The service account must have the following permissions:
+
+* Create, list, watch verbs for all resources that are a part of the install (cannot be scoped to specific resource names).
+  - Permissions to create and manage CustomResourceDefinitions
+  - Permissions to create any other manifest objects
+  - Rules to manage any other cluster-scoped resources
+  - All the rules defined in the CSV under `.spec.install.clusterPermissions` and `.spec.install.permissions`
+  - Permissions for namespace-scoped resources.
+  - These are specified with ClusterRole + ClusterRoleBinding, ClusterRole + RoleBinding, or Role + RoleBinding.
+
+* Get, update, patch, delete verbs for all resources that are a part of the install (can be scoped to specific resource names)
+  - Permissions for cluster-scoped resources
+  - Permissions for namespace-scoped resources
+  - These are created using ClusterRole + ClusterRoleBinding, ClusterRole + RoleBinding, or Role + RoleBinding
+
+* Permissions to create any ClusterRole, ClusterRoleBinding, Role, RoleBinding resources required by the extension.
+  - All the same rules defined in the ClusterRole and Role resources
+  - Escalate and bind verbs for ClusterRole, ClusterRoleBinding, Role, RoleBinding resources
+  - Permissions to create the necessary roles and rolebindings for the controller to be able to perform its job
+
+* Update finalizers on the ClusterExtension to be able to set blockOwnerDeletion and ownerReferences
+
+* Permissions to create the controller deployment, this corresponds to the rules to manage the
+  deployment defined in the ClusterServiceVersion
+
+### Derive minimal RBAC
```

Comment:
Maybe call this section "example" or something like that?

Reply:
I think it could be worthwhile to start from the bundle, list the resources contained within it (and their scope), and list the pertinent parts of the CSV (.clusterPermissions and .permissions).

Then go step-by-step for creating the ClusterRole, and step-by-step for the Role.